packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -0,0 +1,930 @@
+
+  /// The overall shape of a date picker dialog.
+  ///
+  /// {@macro flutter.material.material.shape}
+  final ShapeBorder? shape;
+
+  /// The header's background fill color.
+  final Color? headerBackgroundColor;
+
+  /// The color used for text labels and icons in the header of a date
+  /// picker dialog.
+  ///
+  /// This is used instead of any colors provided by [headerHeadlineStyle]
+  /// or [headerHelpStyle].
+  final Color? headerForegroundColor;
+
+  /// The text style used for the headline text of the header.
+  ///
+  /// The headline text is usually the currently selected date in the header.
+  ///
+  /// The color of the [headerHeadlineStyle] is not used directly, as
+  /// [headerForegroundColor] is used instead.
+  final TextStyle? headerHeadlineStyle;
+
+  /// The text style used for the help text of the header.
+  ///
+  /// The help text (also referred to as "supporting text" in the Material
+  /// spec) is usually a prompt to the user at the top of the header
+  /// (i.e. 'Select date').
+  ///
+  /// The color of the [headerHelpStyle] is not used directly, as
+  /// [headerForegroundColor] is used instead.
+  ///
+  /// See also:
+  ///   [DatePickerDialog.helpText], which specifies the help text.
+  final TextStyle? headerHelpStyle;
+
+  /// The text style used for the row of week day labels at the top
+  /// of the date picker grid.
+  final TextStyle? weekdayStyle;
+
+  /// The text style used for each individual day label in the grid of the
+  /// date picker.
+  ///
+  /// The color of the [dayStyle] is not used directly, as
+  /// [dayForegroundColor] is used instead.
+  final TextStyle? dayStyle;
+
+  /// The color used to paint the day labels in the grid of the date picker.
+  ///
+  /// This will be used instead of the color provided in [dayStyle].
+  final MaterialStateProperty<Color?>? dayForegroundColor;
+
+  /// The color used to paint the background of the day labels in the grid
+  /// of the date picker.
+  final MaterialStateProperty<Color?>? dayBackgroundColor;
+
+  /// The highlight color that's typically used to indicate that
+  /// a day in the grid is focused, hovered, or pressed.
+  final MaterialStateProperty<Color?>? dayOverlayColor;
+
+  /// The color used to paint the [DatePickerDialog.currentDate] label in
+  /// the grid of the date picker.
+  ///
+  /// This will be used instead of the color provided in [dayStyle].
+  final MaterialStateProperty<Color?>? todayForegroundColor;
+
+  /// The color used to paint the background of the
+  /// [DatePickerDialog.currentDate] label in the grid of the date picker.
+  final MaterialStateProperty<Color?>? todayBackgroundColor;
+
+  /// The border used to paint the [DatePickerDialog.currentDate] label in
+  /// the grid of the date picker.
+  final BorderSide? todayBorder;
+
+  /// The text style used to paint each of the year entries in the year
+  /// selector of the date picker.
+  ///
+  /// The color of the [yearStyle] is not used directly, as
```

Comment:
Same comment about deprecation.

Reply:
Addressed in https://github.com/flutter/flutter/pull/119033/commits/024dad5d17523874a83ad6e336b52a9f3f9ea026